linkerd/app/inbound/src/policy/store.rs

```diff
@@ -31,51 +27,46 @@ struct PortHasher(u16);
 
 // === impl Store ===
 
-impl Store {
-    fn mk_default(default: DefaultPolicy) -> Option<(Tx, Rx)> {
-        match default {
-            DefaultPolicy::Deny => None,
-            DefaultPolicy::Allow(sp) => Some(watch::channel(sp)),
-        }
-    }
-
-    pub(crate) fn fixed(
-        default: impl Into<DefaultPolicy>,
+impl<S> Store<S> {
+    pub(crate) fn spawn_fixed(
+        default: DefaultPolicy,
         ports: impl IntoIterator<Item = (u16, ServerPolicy)>,
-    ) -> (Self, Option<Tx>) {
-        let rxs = ports
-            .into_iter()
-            .map(|(p, s)| {
+    ) -> Self {
+        let cache = {
+            let rxs = ports.into_iter().map(|(p, s)| {
                 // When using a fixed policy, we don't need to watch for changes. It's
                 // safe to discard the sender, as the receiver will continue to let us
                 // borrow/clone each fixed policy.
                 let (_, rx) = watch::channel(s);
                 (p, rx)
-            })
-            .collect();
-
-        let (default_tx, default) = match Self::mk_default(default.into()) {
-            Some((tx, rx)) => (Some(tx), Some(rx)),
-            None => (None, None),
+            });
+            Cache::with_permanent_from_iter(Duration::MAX, rxs)
         };
 
-        let store = Self {
-            default,
-            ports: Arc::new(rxs),
-        };
-        (store, default_tx)
+        let (default_tx, default_rx) = watch::channel(ServerPolicy::from(default));
+        tokio::spawn(async move {
+            default_tx.closed().await;
+        });
+
+        Self {
+            cache,
+            default_rx,
+            discover: None,
+        }
     }
 
     /// Spawns a watch for each of the given ports.
     ///
-    /// The returned future completes when a watch has been successfully created for all of the
-    /// provided ports. The store maintains these watches so that each time a policy is checked, it
-    /// may obtain the latest policy provided by the watch. An error is returned if any of the
-    /// watches cannot be established.
-    pub(super) fn spawn_discover<S>(
+    /// The returned future completes when a watch has been successfully created
+    /// for all of the provided ports. The store maintains these watches so that
+    /// each time a policy is checked, it may obtain the latest policy provided
+    /// by the watch. An error is returned if any of the watches cannot be
+    /// established.
```

Comment:
i don't think this comment is correct --- the function returns a `Store`, which is not a future, and a separate watch task is spawned for each provided port. should this be updated?